.github/workflows/linux.yml

```diff
@@ -0,0 +1,141 @@
+name: Build Linux and openSBI in os-boot
+permissions:
+  id-token: write
+  attestations: write
+  contents: read
+
+on:
+  push:
+    paths:
+      - ".github/workflows/linux.yml"
+      - "os-boot/linux/**"
+      - "!os-boot/**.md"
+  pull_request:
+    paths:
+      - ".github/workflows/linux.yml"
+      - "os-boot/linux/**"
+      - "!os-boot/**.md"
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install packages
+        run: sudo apt install -y build-essential curl device-tree-compiler ccache
+
+      - name: Check out Repository Code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 5
+
+      - name: Restore Cache Linux Sources
+        uses: actions/cache/restore@v4
+        id: linux-cache
+        with:
+          key: os-boot-linux-${{ hashFiles('os-boot/linux/Makefile', 'os-boot/linux/*.url') }}
+          path: |
+            os-boot/linux/downloads/
+            os-boot/linux/build/
+
+      - name: Make ELF
+        if: steps.linux-cache.outputs.cache-hit != 'true'
+        working-directory: os-boot/linux
+        run: make -j$(nproc)
+
+      - name: Save Cache Linux Sources
+        if: steps.linux-cache.outputs.cache-hit != 'true'
+        uses: actions/cache/save@v4
+        with:
+          key: ${{ hashFiles('os-boot/linux/Makefile', 'os-boot/linux/*.url') }}
+          path: |
+            os-boot/linux/downloads/
+            os-boot/linux/build/
+
+      - name: Make dtb
+        working-directory: os-boot/linux
+        run: make -j$(nproc) build/sail.dtb
```

Comment:
This needs to happen after building sail now that we generate the .dts from the sail simulator binary.